packages/host/app/components/operator-mode/code-submode/playground-panel.gts

```diff
@@ -0,0 +1,169 @@
+import type { TemplateOnlyComponent } from '@ember/component/template-only';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { cached, tracked } from '@glimmer/tracking';
+
+import { LoadingIndicator, BoxelSelect } from '@cardstack/boxel-ui/components';
+
+import { getCards, type ResolvedCodeRef } from '@cardstack/runtime-common';
+
+import { getCodeRef, type CardType } from '@cardstack/host/resources/card-type';
+import { ModuleContentsResource } from '@cardstack/host/resources/module-contents';
+
+import type RealmServerService from '@cardstack/host/services/realm-server';
+
+import type { CardDef } from 'https://cardstack.com/base/card-api';
+
+const getItemTitle = (item: CardDef, displayName?: string) => {
+  if (!item) {
+    return;
+  }
+  if (item.title) {
+    return item.title;
+  }
+  let fallbackName = displayName ?? item.constructor.displayName ?? 'Card';
+  return `Untitled ${fallbackName}`;
+};
+
+const SelectedItem: TemplateOnlyComponent<{ Args: { title?: string } }> =
+  <template>
+    <div class='selected-item'>
+      Instance:
+      <span class='title' data-test-selected-item>
+        {{@title}}
+      </span>
+    </div>
+    <style scoped>
+      .selected-item {
+        display: flex;
+        align-items: center;
+        gap: var(--boxel-sp-xs);
+        overflow: hidden;
+        font: 600 var(--boxel-font-xs);
+        letter-spacing: var(--boxel-lsp-sm);
+      }
+      .title {
+        display: -webkit-box;
+        -webkit-box-orient: vertical;
+        -webkit-line-clamp: 2;
+      }
+    </style>
+  </template>;
+
+interface PlaygroundContentSignature {
+  Args: {
+    codeRef: ResolvedCodeRef;
+    displayName?: string;
+  };
+}
+class PlaygroundPanelContent extends Component<PlaygroundContentSignature> {
+  <template>
+    <BoxelSelect
+      class='instance-chooser'
+      @options={{this.instances}}
+      @selected={{this.selectedItem}}
+      @selectedItemComponent={{if
+        this.selectedItem
+        (component
+          SelectedItem title=(getItemTitle this.selectedItem @displayName)
+        )
+      }}
+      @onChange={{this.onSelect}}
+      @placeholder='Please Select'
+      data-test-instance-chooser
+      as |item|
+    >
+      {{getItemTitle item @displayName}}
+    </BoxelSelect>
+    <style scoped>
+      .instance-chooser {
+        color: var(--boxel-dark);
+        height: var(--boxel-form-control-height);
+      }
+    </style>
+  </template>
+
+  @service private declare realmServer: RealmServerService;
+  @tracked private selectedItem?: CardDef;
+
+  private options = getCards(
+    () => ({
+      filter: { type: this.args.codeRef },
+      sort: [{ by: 'createdAt', direction: 'desc' }],
+    }),
+    () => this.realmServer.availableRealmURLs,
+  );
```

Comment:
I was imagining this would be based on the recent files list in code mode (most recently viewed). But I like using this query. Maybe we should combine the ideas and run the query, then bubble up items on the recently accessed list to the top? Also, should have a max that we show in this list -- maybe 5?